### DISCUSSION TITLE
A way to show what object in an array in the document matched the query.

### DISCUSSION THREAD

**Discussion by labadcloyd** [meilisearch/product#655] [Feedback & Feature Proposal]
I have a document in an index like so:

```json
{
            "id": "9ef0e1cb-0c15-4cf0-a655-6e83af162bcb",
            "ingredient_mapping": [
                {
                    "ingredient_name": "Croissants",
                    "ingredient_variant_Name": "cheese",
                },
                { 
                    "ingredient_name": "Croissants", 
                    "ingredient_variant_Name": "butter", 
                },
                { 
                    "ingredient_name": "Croissants",
                    "ingredient_variant_Name": "apple", 
                }
            ],
            "name": "Croissants",
            "name_owner": "USDA"
}
```
 
If im going to search for the term "Croissants butter" it would most likely return this entire object. But if there was a way to know at which index of the array "ingredient_mapping" the match was then it would really help a lot. 

This is similar to showMatchesPosition but the index of the object in the array would be returned, not the index of the string.

EDIT: I've queried the above document with showMatcherPosition and this is what it returns:
```json
{
  "hits": [
    {
      "_matchesPosition": {
        "ingredient_details.ingredient_variant_name": [
          {
            "length": 6,
            "start": 0
          }
        ],
        "name": [
          {
            "length": 10,
            "start": 0
          }
        ]
      },
      "id": 129,
      "ingredient_details": [
        {
          "ingredient_mapping_id": 1950,
          "ingredient_subvariant_name": "",
          "ingredient_variant_name": "cheese"
        },
        {
          "ingredient_mapping_id": 1951,
          "ingredient_subvariant_name": "",
          "ingredient_variant_name": "butter"
        },
        {
          "ingredient_mapping_id": 1952,
          "ingredient_subvariant_name": "",
          "ingredient_variant_name": "apple"
        }
      ],
      "name": "Croissants",
      "name_owner": "USDA"
    }
  ],
  "estimatedTotalHits": 1,
  "limit": 20,
  "processingTimeMs": 1,
  "query": "croissant butter"
}
```

Is  there a way we can add the index to the array in the "_matchesPosition" field data?

**Comment by macraig** [comment]
Hi @labadcloyd , thanks for taking the time to share your use case!

We have an ongoing discussion for this need in https://github.com/meilisearch/product/discussions/550, I'll move your feedback there and close this discussion to unify the source of truth. Feel free to add to that conversation and/or upvote the feature on [Productboard](https://meilisearch.productboard.com/portal/1-meilisearch/tabs/610f17a9-3989-4418-b4ee-16a65b102e41/features/20669965/portal/expanded).
